File: backend/app/schemas/selection.py
from pydantic import BaseModel


class StockScoreSchema(BaseModel):
    rank: int
    ts_code: str
    name: str | None = None
    industry: str | None = None
    composite_score: float
    status: str = "OK"
    close: float | None = None
    pct_change: float | None = None
    pe_ttm: float | None = None
    pb: float | None = None
    roe_ttm: float | None = None
    market_cap: float | None = None
    factor_snapshot: dict[str, float] = {}


class SelectionResultSchema(BaseModel):
    trade_date: str
    strategy_name: str
    universe_count: int
    filtered_count: int
    candidate_count: int
    final_count: int
    results: list[StockScoreSchema] = []

File: frontend/src/types/selection.ts
export interface StockScore {
  rank: number
  ts_code: string
  name?: string | null
  industry?: string | null
  composite_score: number
  status: string
  close?: number | null
  pct_change?: number | null
  pe_ttm?: number | null
  pb?: number | null
  roe_ttm?: number | null
  market_cap?: number | null
  factor_snapshot: Record<string, number>
}

export interface SelectionResult {
  trade_date: string
  strategy_name: string
  universe_count: number
  filtered_count: number
  candidate_count: number
  final_count: number
  results: StockScore[]
}

export interface SelectionRunRequest {
  strategy_id?: string | null
  trade_date?: string | null
}

export interface SelectionRecord {
  strategy_id: string
  ts_code: string
  trade_date: string
  rank: number
  composite_score: number
  status: string
  factor_snapshot: Record<string, number>
  created_at?: string | null
}
